pyphare/pyphare/pharesee/particles.py

```diff
@@ -0,0 +1,40 @@
+
+import numpy as np
+
+class Particles():
+    def __init__(self, **kwargs):
+        if "box" in kwargs:
+            box = kwargs["box"]
+            self.iCells = np.random.randint(box.lower, high=box.upper+1, size=box.size()*100)
+            self.deltas = np.random.rand(box.size()*100)
+            self.vx = np.random.randn(box.size()*100)
+            self.vz = np.random.randn(box.size()*100)
+            self.vy = np.random.randn(box.size()*100)
+        else:
+            self.iCells = kwargs["icell"]
+            self.deltas = kwargs["deltas"]
+            self.vx = kwargs["vx"]
+            self.vy = kwargs["vy"]
+            self.vz = kwargs["vz"]
+
+    def select(self, box):
+        idx = np.where((self.iCells >= box.lower) & (self.iCells <= box.upper))[0]
+        return Particles(icell=self.iCells[idx],
+                         deltas=self.deltas[idx],
+                         vx=self.vx[idx],
+                         vy=self.vy[idx],
+                         vz=self.vz[idx])
+
+
+    def add(self, particles):
```

Comment:
where is this used?

Reply:
https://github.com/PHAREHUB/PHARE/pull/184/files?file-filters%5B%5D=.py&hide-deleted-files=true#diff-cecd1154f41b00ad187aba0195960bbcR100-R101

Reply:
doesn't seem to be jumping to where I guess you have linked

Reply:
https://github.com/nicolasaunai/PHARE/blob/python/pyphare/tests/test_pharesee/test_geometry.py#L100